executable/microsoft_pe.ksy

```diff
@@ -408,3 +414,144 @@ types:
         size: length - 8
         doc: Contains a certificate, such as an Authenticode signature.
     doc-ref: 'https://docs.microsoft.com/en-us/windows/desktop/debug/pe-format#the-attribute-certificate-table-image-only'
+
+  resource_directory_table:
+    doc-ref: 'https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#resource-directory-table'
+    params:
+      - id: depth
+        type: u4
+      - id: section_file_offset
+        type: u4
+      - id: section_virtual_address
+        type: u4
+    seq:
+      - id: reserved
+        -orig-id: characteristics
+        type: u4
+        doc: Reserved for future use, should be set to zero
+      - id: timestamp
+        type: u4
+        -orig-id: TimeDateStamp
+      - id: version
+        type: version_u2
+      - id: num_named_entries
+        -orig-id: NumberOfNamedEntries
+        type: u2
+      - id: num_id_entries
+        -orig-id: NumberOfIdEntries
+        type: u2
+      - id: named_entries
+        type: resource_directory_entry(depth)
+        repeat: expr
+        repeat-expr: num_named_entries
+      - id: id_entries
+        type: resource_directory_entry(depth)
+        repeat: expr
+        repeat-expr: num_id_entries
+
+  resource_directory_entry:
+    doc-ref: 'https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#resource-directory-entries'
+    params:
+      - id: depth
+        type: u4
+    seq:
+      - id: resource_type
+        type: u4
+        enum: enum_resource_type
+        if: depth == 0
+      - id: ofs_name
```

Comment:
How do you suggest naming this field that can be either and `id` or an offset to a string (which is a `name`), depending on the most-significant bit (`is_name`)?

Reply:
The best thing to do would be to have two separate fields switched using `if`:

```ksy
  - id: ofs_name
    type: b31
    if: ... 
  - id: some_id
    type: not ... 
```

But that's not possible of course, because the `is_name` comes after these fields.

Regarding the name, I'd